### PR TITLE
deduplicate dns zones passed to operator

### PR DESCRIPTION
**Pull request by OliverMKing**
# Description

Deduplicates any dns zones passed to the operator. Will need similar logic of some form in RP too.

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Locally, e2e, unit

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by OliverMKing** [comment]
failing test coverage check because I added a unit test to utils which had no coverage before. should add more unit tests to utils later

**Review by bosesuneha — APPROVED** [review]
lgtm

File: pkg/controller/dns/external_dns_test.go [hunk 1]
```diff
@@ -98,7 +98,7 @@ func TestPublicConfig(t *testing.T) {
 				Subscription:       allZones.PublicZoneConfig.Subscription,
 				ResourceGroup:      allZones.PublicZoneConfig.ResourceGroup,
 				Provider:           manifests.PublicProvider,
-				DnsZoneResourceIDs: allZones.PublicZoneConfig.ZoneIds,
+				DnsZoneResourceIDs: util.Keys(allZones.PublicZoneConfig.ZoneIds),
```
Comment (jaiveerk) [inline pkg/controller/dns/external_dns_test.go:101]: keys in maps in golang aren't ordered right? so this has the potential to be flaky?

wonder if sorting the keys function in utils is the best way to remedy inconsistency

**Review by jaiveerk — APPROVED** [review]
minor test nit but lgtm

**Pull request merged**